gammapy/cube/core.py

```diff
@@ -186,10 +190,11 @@ def fill_events(self, events, weights=None):
         """
         if weights is not None:
             weights = events[weights]
+        xx, yy = self.sky_image_ref.wcs_skycoord_to_pixel(events.radec)
+        zz = events.energy
+        bins = self.energy_axis.energy, self.sky_image_ref._bins_pix[0], \
```

Comment:
Can you replace `self.energy_axis.energy` by a call to `self.energies(mode='edges')`?